packages/core/src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-alternative-methods-button-strategy.ts

```diff
@@ -0,0 +1,143 @@
+import { FormPoster } from '@bigcommerce/form-poster';
+
+import { CheckoutActionCreator, CheckoutStore } from '../../../checkout';
+import { InvalidArgumentError, MissingDataError, MissingDataErrorType, } from '../../../common/error/errors';
+import { PaymentMethod } from '../../../payment';
+import { PaymentMethodClientUnavailableError } from '../../../payment/errors';
+import { ApproveDataOptions, ButtonsOptions, PaypalButtonStyleOptions, PaypalCommerceRequestSender, PaypalCommerceScriptLoader, PaypalCommerceSDK } from '../../../payment/strategies/paypal-commerce';
+import { CheckoutButtonInitializeOptions } from '../../checkout-button-options';
+import CheckoutButtonStrategy from '../checkout-button-strategy';
+
+import getValidButtonStyle from './get-valid-button-style';
+
+export default class PaypalCommerceAlternativeMethodsButtonStrategy implements CheckoutButtonStrategy {
+    private _paypalCommerceSdk?: PaypalCommerceSDK;
+    private _paymentMethod?: PaymentMethod;
+
+    constructor(
+        private _store: CheckoutStore,
+        private _checkoutActionCreator: CheckoutActionCreator,
+        private _formPoster: FormPoster,
+        private _paypalScriptLoader: PaypalCommerceScriptLoader,
+        private _paypalCommerceRequestSender: PaypalCommerceRequestSender
+    ) {}
+
+    async initialize(options: CheckoutButtonInitializeOptions): Promise<void> {
+        const { paypalcommercealternativemethods, containerId, methodId } = options;
+        const { apm, initializesOnCheckoutPage, style } = paypalcommercealternativemethods || {};
+
+        if (!methodId) {
+            throw new InvalidArgumentError('Unable to initialize payment because "options.methodId" argument is not provided.');
+        }
+
+        if (!containerId) {
+            throw new InvalidArgumentError(`Unable to initialize payment because "options.containerId" argument is not provided.`);
+        }
+
+        if (!paypalcommercealternativemethods) {
+            throw new InvalidArgumentError(`Unable to initialize payment because "options.paypalcommercealternativemethods" argument is not provided.`);
+        }
+
+        if (!apm) {
+            throw new InvalidArgumentError(`Unable to initialize payment because "options.paypalcommercealternativemethods.apm" argument is not provided.`);
+        }
+
+        const state = await this._store.dispatch(this._checkoutActionCreator.loadDefaultCheckout());
+        const currency = state.cart.getCartOrThrow().currency.code;
+        this._paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(methodId);
+        this._paypalCommerceSdk = await this._paypalScriptLoader.loadPaypalCommerce(this._paymentMethod, currency, initializesOnCheckoutPage);
```

Comment:
What's the initializesOnCheckoutPage flag?

Reply:
We are using this flag to understand is the button strategy initialize on Checkout page or not. We should change `commit` paypal sdk script loader option to different boolean values on checkout or on cart page

Reply:
If we set commit true, we can use shipping option or inline checkout functional (we are working on them), and we can see a bunch of APMs what we should not show to customer. A bit difficult logic there, but it's specific only for paypal commerce provider